tests/test_testing.py

```diff
@@ -449,7 +449,7 @@ def test_isolation_stderr_errors():
     with runner.isolation() as (_, err, _):
         click.echo("\udce2", err=True, nl=False)
 
-    assert err.getvalue() == b"\\udce2"
```

Comment:
With my patch, stream objects yielded from `runner.isolation`, including `err`, may be closed as the context manager exits, dereferencing (hence, allowing gc to collect) its internal `StreamMixer` reference. I believe this race is already possible in click 8.2.1, albeit much less likely, so I'm hoping that this change is acceptable.